app/domain/authentication/authn_k8s/set_file_content_in_container.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'command_class'
+
+module Authentication
+  module AuthnK8s
+
+    SetFileContentInContainer ||= CommandClass.new(
+      dependencies: {
+        kubectl_exec:      KubectlExec.new,
+        k8s_object_lookup: K8sObjectLookup,
+        logger:            Rails.logger
+      },
+      inputs: %i(webservice pod_namespace pod_name path content mode container)
+    ) do
+
+      LOG_FILE = "${TMPDIR:-/tmp}/conjur_set_file_content.log"
+
+      def call
+        set_file_content_in_container
+      end
+
+      private
+
+      def set_file_content_in_container
+        @kubectl_exec.call(
+          k8s_object_lookup: @k8s_object_lookup.new(@webservice),
+          pod_namespace:     @pod_namespace,
+          pod_name:          @pod_name,
+          container:         @container,
+          cmds:              %w(sh),
+          body:              set_file_content_script(@path, @content, @mode),
+          stdin:             true
+        )
+      end
+
+      # Sets the content of a file in a given path to the given content
+      # We first copy the content into a temporary file and only then move it to
+      # the desired path as the client polls on its existence and we want it to
+      # exist only when the whole content is present.
+      #
+      # We redirect the output to a log file on the authn-client container
+      # that will be written in its logs for supportability.
+      def set_file_content_script(path, content, mode)
+        tmp_cert = "#{path}.tmp"
+
+        "
+#!/bin/sh
```

Comment:
Need to verify that the file was created well with the relevant inputs and permissions

Reply:
wdym? 

1. verify where? 
2. what is "created well with the relevant inputs and permissions"
3. which file? the log file or cert file?

Reply:
1. Some unit test which verify that the file exist in the relevant path with the relevant permissions
2. Since you declare the path and the content need to verify it created in the relevant path with the relevant content
3. cert tile

Reply:
1. It can't be a UT because this logic happens in kubectl exec so there's nothing to UT here.
2 + 3. we verify that the cert is injected to the container already